docker/rucio_client/scripts/cmslinks.py

```diff
@@ -72,9 +73,19 @@ def _get_rselist(self, rselist=None):
                 except Exception as e:
                     logging.warning(f'No PNN for RSE {rse}. Trying to get it from gitlab. Error: {str(e)}')
                 for site in sites:
-                    if site.get('rse') in rse:
-                        pnn = site.get('site')
-                        break
+                    # Error handling in case there are issues retrieving info from sites' dicts
+                    try:
+                        if site.get('rse') in rse:
+                            pnn = site.get('site')
+                            break
+                    except Exception as e:
+                        logging.warning(f'Problem getting PNN from gitlab for RSE {rse} SITE {site.get("site")}. Error: {str(e)}')
+            
+            # If PNN could be retrieved skip to the next RSE (do not add entry to the list)
+            if pnn is None:
```

Comment:
You can add the continue under exceptions above.

Reply:
The problem here is that the desired 'rse' field may be empty. Moving `continue` under the exception above will skip to the next site. However, in case that no sites have the 'rse' field set, the pnn value will remain 'None'. Appending this value to `self.rselist` will create problems in other parts of the code. So, preventing RSEs with pnn value of None to be appended in the list is mandatory.